utils/codec/codec.go

```diff
@@ -259,26 +236,21 @@ func (c *codec) marshal(value reflect.Value, p *wrappers.Packer) error {
 // Unmarshal unmarshals [bytes] into [dest], where
 // [dest] must be a pointer or interface
 func (c *codec) Unmarshal(bytes []byte, dest interface{}) error {
+	if dest == nil {
+		return errUnmarshalNil
+	}
+
 	c.lock.Lock()
 	defer c.lock.Unlock()
```

Comment:
It would be nice if we could change this to be a read lock...the only place inside `unmarshal` that we modify the codec's state is in `getSerializedFieldIndices`...perhaps we could just register a callback there instead of writing